account_invoice_overdue_reminder/models/account_invoice_overdue_reminder.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2020 Akretion France (http://www.akretion.com/)
+# @author: Alexis de Lattre <alexis.delattre@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class AccountInvoiceOverdueReminder(models.Model):
+    _name = 'account.invoice.overdue.reminder'
+    _description = 'Overdue Invoice Reminder Action History'
+    _order = 'id desc'
+
+    # For the link to invoice: why a M2O and not a M2M ?
+    # Because of the "counter" field: a single reminder action for a customer,
+    # the "counter" may not be the same for each invoice
+    invoice_id = fields.Many2one(
+        'account.invoice', string='Invoice', ondelete='cascade', readonly=True)
+    action_id = fields.Many2one(
+        'overdue.reminder.action', string='Overdue Reminder Action',
+        ondelete='cascade')
+    action_commercial_partner_id = fields.Many2one(
+        related='action_id.commercial_partner_id', store=True)
+    action_partner_id = fields.Many2one(
+        related='action_id.partner_id', store=True)
+    action_date = fields.Date(related='action_id.date', store=True)
+    action_user_id = fields.Many2one(related='action_id.user_id')
+    action_reminder_type = fields.Selection(
+        related='action_id.reminder_type', store=True)
+    action_result_id = fields.Many2one(
+        related='action_id.result_id', readonly=False)
+    action_result_notes = fields.Text(
+        related='action_id.result_notes', readonly=False)
+    action_mail_id = fields.Many2one(
+        related='action_id.mail_id')
+    action_mail_state = fields.Selection(
+        related='action_id.mail_id.state', string='E-mail Status')
+    counter = fields.Integer(readonly=True)
+    company_id = fields.Many2one(
+        related='invoice_id.company_id', store=True)
+
+    _sql_constraints = [(
+        'counter_positive',
+        'CHECK(counter >= 0)',
+        'Counter must always be positive')]
+
+    @api.constrains('invoice_id')
+    def invoice_id_check(self):
+        for action in self:
+            if action.invoice_id and action.invoice_id.type != 'out_invoice':
+                raise ValidationError(_(
+                    "An overdue reminder can only be attached "
+                    "to a customer invoice"))
+
+    @api.depends('invoice_id', 'counter')
+    def name_get(self):
+        res = []
+        for rec in self:
+            name = _('%s Reminder %d') % (rec.invoice_id.number, rec.counter)
```

Comment:
```suggestion
            name = _("{} Reminder {}".format(rec.invoice_id.number, rec.counter))
```
I think a new format is better. What do you think about it?

Reply:
What is the real advantage of it ? The result is the same, no ?

Reply:
In this place, the result is the same but the new format string syntax has become more powerful without complicating the simpler use cases. You can see other reasons in https://realpython.com/python-string-formatting/#which-string-formatting-method-should-you-use.

Reply:
Thanks to inform us about string formatting. As it's python 3.6+ I wonder if it works like that
`name = _(f'{rec.invoice_id.number} Reminder {rec.counter}')`

If yes the syntax is really readable